pkg/api/controllers/fileshare.go

```diff
@@ -549,6 +549,22 @@ func (f *FileShareSnapshotPortal) CreateFileShareSnapshot() {
 		return
 	}
 
+	// Check existence of fileshare snapshot name #931
+	filesnaps, err := db.C.ListFileShareSnapshots(ctx)
+	if err != nil{
+		errMsg := fmt.Sprintf("get list of fileshare snapshot failed: %s", err.Error())
+		f.ErrorHandle(model.ErrorBadRequest, errMsg)
+		return
+	}else{
+		for _, filesnap := range filesnaps {
+			if filesnap.Name == snapshot.Name {
```

Comment:
snapshot name is case sensitive, right?

Reply:
No it is not,  but some naming standards are there like. It should not start with numbers and special characters. This condition is to check the name already exists in db or not.